src/io/simulator/simulator_stats.hpp

```diff
@@ -21,5 +25,22 @@ struct SimulatorStats {
   uint64_t total_requests = 0;
   uint64_t total_responses = 0;
   uint64_t simulator_ticks = 0;
+  Duration elapsed_time;
+
+  friend bool operator==(const SimulatorStats & /* lhs */, const SimulatorStats & /* rhs */) = default;
+
+  friend std::ostream &operator<<(std::ostream &in, const SimulatorStats &stats) {
+    auto elapsed_ms = stats.elapsed_time.count() / 1000;
```

Comment:
As the type of `Duration` can be easily changed from microseconds, please use `std::chrono::duration_cast<std::chrono::milliseconds>` to make sure we don't break this.

Reply:
thanks, applied